pkg/db/src/implementations/version-1.2/customer-orders.ts

```diff
@@ -0,0 +1,168 @@
+import { BehaviorSubject, firstValueFrom, Observable, ReplaySubject, share, Subject } from "rxjs";
+
+import { VolumeStock } from "@librocco/shared";
+
+import { DocType } from "@/enums";
+
+import { VersionedString, OrdersDatabaseInterface, CustomerOrderInterface, CustomerOrderData } from "@/types";
+
+import { versionId } from "./utils";
+import { isEmpty, isVersioned, runAfterCondition, uniqueTimestamp } from "@/utils/misc";
+import { newDocumentStream } from "@/utils/pouchdb";
+
+
+class CustomerOrder implements CustomerOrderInterface {
+
+	// {
+	// 	"id": string, // use 'timestampedId' (or 'timestampId', not sure, it's in the 'db' package, under utils I think
+	// 	"draft": boolean,
+	// 	"displayName": string, // "name" -> "displayName" to keep consistent with the rest of the app
+	// 	"email": string,
+	// 	"books": [ 
+	// 	  { isbn: string }, // No status for now we can add that later when dealing with supplier orders
+	// 	],
+	// 	"deposit"?: number // I think the deposit should be calculated on display without persisting and then added when the order is placed
+	// 	}
+
+	#db: OrdersDatabaseInterface;
+
+	#initialized = new BehaviorSubject(false);
+	#exists = false;
+
+	draft = true;
+
+	orderId = "";
+	email = "";
+	deposit = 0;
+
+	books = []
+
+	// Update stream receives the latest document (update) stream from the db. It's multicasted using plain RxJS Subject (no repeat or anything).
+	// This stream is used to signal the update has happened (and has been streamed to update the instance). Subscribers needing to be notified when
+	// an update happens should subscribe to this stream.
+	#updateStream: Observable<CustomerOrderData>;
+	// The stream is piped from the update stream, only it's multicasted using a ReplaySubject, which will cache the last value emitted by the stream,
+	// for all new subscribers. Subscribers needing the latest (up-to-date) data and not needing to be notified when the NEXT update happened, should
+	// subscribe to this stream.
+	#stream: Observable<CustomerOrderData>;
+
+	// timestramped uid
+	_id: VersionedString;
+	docType = DocType.CustomerOrder;
+	_rev?: string;
+	_deleted?: boolean;
+
+
+	entries: VolumeStock[] = [];
+	committed = false;
+	displayName = "";
+	updatedAt: string | null = null;
+
+	constructor(db: OrdersDatabaseInterface, id?: string) {
+		this.#db = db;
+
+		// Store the id internally:
+		// - if id is a single segment id, prepend the warehouse id and version the string
+		// - if id is a full id, assign it as is
+		this._id = !id
+			? versionId(`${uniqueTimestamp()}`)
+			: isVersioned(id, "v1") // If id is versioned, it's a full id, assign it as is
+				? id
+				: versionId(`${id}`);
```

Comment:
This can be replaced with `versionId(id)` - as it's idempotent in terms: It won't prepend the already versioned `id`.
Example
```ts
versionId("order-1") // returns "v1/order-1"
versionId("v1/order-1") // returns "v1/order-1"
```

Reply:
Also, keep in mind that, during the development, the two dbs should be ran in isolation (orders and inventory), because: If we were to write to the same db using both interfaces, we would get interference:
- warehouse id (path): `v1/<warehouse-id>`
- customer order id (path): `v1/<order-id>`

This is fine during development, later we'll namespace the different sections, and I've opened up a ticket for that (just pointing this out to be kept in mind)